src/test/java/org/gridsuite/directory/server/DirectoryElementRepositoryTest.java

```diff
@@ -52,4 +52,35 @@ void testFindAllByIdInAndParentIdAndTypeNot() {
 
         assertThat(expectedResult).usingRecursiveComparison().ignoringCollectionOrder().isEqualTo(directoryElementRepository.findAllByIdInAndParentIdAndTypeNotAndStashed(insertedElement.stream().map(e -> e.getId()).toList(), parentDirectoryUuid, DIRECTORY, false));
     }
+
+    @Test
+    void testCountCasesByUser() {
+        String userId1 = "user1";
+        DirectoryElementEntity parentDirectory = directoryElementRepository.save(
+                createRootElement("root", "DIRECTORY", userId1)
+        );
+        UUID parentDirectoryUuid = parentDirectory.getId();
+
+        List<DirectoryElementEntity> insertedElement = directoryElementRepository.saveAll(List.of(
+                createElement(parentDirectoryUuid, "dir1", "DIRECTORY", userId1),
+                createElement(parentDirectoryUuid, "filter1", "FILTER", userId1),
+                createElement(parentDirectoryUuid, "study1", "STUDY", userId1),
+                createElement(parentDirectoryUuid, "study2", "STUDY", userId1),
+                createElement(UUID.randomUUID(), "studyFromOtherDir", "STUDY", userId1),
+                createElement(parentDirectoryUuid, "case1", "CASE", userId1),
+                createElement(parentDirectoryUuid, "case2", "CASE", userId1),
+                createElement(parentDirectoryUuid, "case3", "CASE", userId1),
+                createElement(UUID.randomUUID(), "case4", "CASE", userId1),
+                createElement(parentDirectoryUuid, "case5", "CASE", "user2"),
+                createElement(UUID.randomUUID(), "case6", "CASE", "user2")
```

Comment:
create another element with type 'STUDY' for user2 since this test does not detect the parentheses error :)